Realm/Realm/DatabaseTypes/Accessors/ManagedAccessor.cs

```diff
@@ -169,6 +171,18 @@ public IRealmObjectBase GetParent()
             return Realm.MakeObject(parentMetadata, parentHandle);
         }
 
+        private IntPtr[] GetPropertyIndices()
+        {
+            if (_propertyIndices != null)
+            {
+                return _propertyIndices;
+            }
+
+            const PropertyType collectionFlag = PropertyType.Array | PropertyType.Dictionary | PropertyType.Set;
+            _propertyIndices = (from p in ObjectSchema where (p.Type & collectionFlag) == 0 select Metadata.GetPropertyIndex(p.Name)).ToArray();
```

Comment:
```suggestion
            _propertyIndices = ObjectSchema.Where(p => !p.Type.HasFlag(collectionFlag).Select(p => Metadata.GetPropertyIndex(p.Name).ToArray();
```
I think this reads clearer, specifically the usage of [Enum.HasFlag](https://learn.microsoft.com/en-us/dotnet/api/system.enum.hasflag).

Reply:
I tried this earlier and didn't seem to succeed, now I also tried to adopt your suggestion but that made tests failed. After reading about it I found this snippet which implies that the semantics are not really what I want. 
<img width="845" alt="image" src="https://user-images.githubusercontent.com/44238293/208046025-58bbf1ac-bfe1-414c-b3ff-e5337ee10acf.png">

if I know my logic that would imply that I have to write something in terms of 

`(p.Type.HasFlag(PropertyType.Array) || p.Type.HasFlag(PropertyType.Dictionary || p.Type.HasFlag(PropertyType.Set)))`

which looks slightly bloated IMO..

Reply:
You're right, my usage of `HasFlag` is not correct here. Luckily I found out we have an [`IsCollection()` extension method on `PropertyType`](https://github.com/realm/realm-dotnet/blob/f7a10a89d0e8ea3cc8d91056e7f63c572f983d6c/Realm/Realm/Schema/PropertyTypeEx.cs#L205-L224), so you should be able to do something like `ObjectSchema.Where(p => !p.Type.IsCollection(out var _))` (we deliberately discard the out parameter because we don't care for it).